docs/output-targets/dist.md

```diff
@@ -73,6 +73,12 @@ import * as utils from '../path/to/utils';
 If using the `dist-collection` output target directly, the same result can be achieved using the [`transformAliasedImportPaths`](../output-targets/dist.md#transformaliasedimportpathsincollection) flag on the target's config.
 :::
 
+### esmLoaderPath
+
+*default: `/dist/loader`*
+
+Provide a custom path for the ESM loader directory, containing files you can import in an initiation script within your application to register all your components for lazy loading.
+
```

Comment:
Can we expand on this? I'm not sure that as a user, I'd know enough here to start using this flag. To start, it might be worthwhile to include an example directory structure, setting with a custom value for `esmLoaderPath`, and explaining how setting `esmLoaderPath` affects the output of Stencil.

Reply:
Agree. Maybe we need an additional section that talks about lazy loading and what the loader does? We could pull some of this out of the "How is this different from "dist-custom-elements" output target?" section.

Reply:
I added a section to the loader script and also added an example how modifying the option has an impact on the user of a stencil library.